code/script.js

```diff
@@ -1,325 +1,452 @@
+  questions.innerHTML = oldQuestions;
+
+  // clearing the log
+  log.innerHTML = "";
+
+  // reset the number of guesses showing
+  numberOfGuesses = 0;
+  updateGuessCounter();
+
+  // showing the game board on the screen
+  generateBoard();
+
+  // randomizing the person you're looking for
+  setSecret();
+};
 
 // setting the currentQuestion object when you select something in the dropdown
 const selectQuestion = () => {
-  const category = questions.options[questions.selectedIndex].parentNode.label
+  const category = questions.options[questions.selectedIndex].parentNode.label;
+  const value = questions.options[questions.selectedIndex].value;
 
   // This variable stores what option group (category) the question belongs to.
   // We also need a variable that stores the actual value of the question we've selected.
-  // const value =
 
   currentQuestion = {
     category: category,
-    // value: value
-  }
-}
+    value: value,
+  };
+};
 
 // This function should be invoked when you click on 'Find Out' button.
 const checkQuestion = () => {
-  const { category, value } = currentQuestion
+  // if the "make a guess" option (it's the first one so it has index 0) is selected, pressing "find out" should not do anything
+  if (questions.options[questions.selectedIndex].index === 0) {
+    return;
+  }
+
+  const { category, value } = currentQuestion;
+  numberOfGuesses++; // update the number of guesses
+  updateGuessCounter(); // and make sure to update the guess counter showing on the page also :)
 
   // Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
   // See if we should keep or remove people based on that
   // Then invoke filterCharacters
-  if (category === 'hair' || category === 'eyes') {
-
-  } else if (category === 'accessories' || category === 'other') {
-
+  if (category === "hair") {
+    if (value === secret.hair) {
+      filterCharacters(true);
+    } else {
+      filterCharacters(false);
+    }
+  } else if (category === "eyes") {
+    if (value === secret.eyes) {
+      filterCharacters(true);
+    } else {
+      filterCharacters(false);
+    }
+  } else if (category === "accessories" || category === "other") {
+    if (secret.accessories.includes(value) || secret.other.includes(value)) {
+      filterCharacters(true);
+    } else {
+      filterCharacters(false);
+    }
   }
-}
+};
 
 // It'll filter the characters array and redraw the game board.
 const filterCharacters = (keep) => {
-  const { category, value } = currentQuestion
+  const { category, value } = currentQuestion;
   // Show the correct alert message for different categories
-  if (category === 'accessories') {
+  if (category === "hair") {
     if (keep) {
-      alert(
-        `Yes, the person wears ${value}! Keep all people that wears ${value}`
-      )
+      addToLog(`The person has ${value} ${category}!`);
```

Comment:
A cool feature with the log, instead of alerts. 
It's not necessary but to clarify what happens in the game
you could add e.g. "remove all people with a hat" or "keep all people with a hat" in the log.